### PR TITLE
refactor `listBuckets`

### PR DESCRIPTION
**Pull request by prakashsvmx**
refactor `listBuckets` to typescript

### CODE REVIEW

File: src/internal/client.ts [hunk 1]
```diff
@@ -864,4 +865,12 @@ export class TypedClient {
     const res = await this.makeRequestAsync({ method, bucketName, objectName, query })
     return xmlParsers.parseListParts(await readAsString(res))
   }
+
+  listBuckets(callback: ResultCallback<BucketItemFromList[]>): Promise<BucketItemFromList[]>
```
Comment (trim21) [inline src/internal/client.ts:869]: This maybe convenient but it doesn't help TypeScript to narrow overload, should split this with a promise api and a callback api

Comment (prakashsvmx) [reply src/internal/client.ts:869]: Could you please elaborate this? is the overload not sufficient ?

Comment (trim21) [reply src/internal/client.ts:869]: `listBuckets(()=>{}).then(result => {}, err => {} )` is not actually valid, but still pass type checking

Comment (trim21) [reply src/internal/client.ts:869]: And the easiest way is always to bump major version and drop callback style api...

Comment (prakashsvmx) [reply src/internal/client.ts]: Yes. Thank you. :+1:

File: examples/list-buckets.js [hunk 2]
```diff
@@ -25,7 +25,12 @@ var s3Client = new Minio.Client({
   secretKey: 'YOUR-SECRETACCESSKEY',
 })
 
-s3Client.listBuckets(function (e, buckets) {
-  if (e) return console.log(e)
-  console.log('buckets :', buckets)
-})
+const listBucketsExample = async () => {
```
Comment (trim21) [inline examples/list-buckets.js:28]: can we just rename this file to mjs?

Comment (trim21) [reply examples/list-buckets.js:28]: also add a global notice about toplevel await， so code snippet can be more focused

File: src/internal/xml-parser.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { XMLParser } from 'fast-xml-parser'
 
 import * as errors from '../errors.ts'
 import type { BucketItemWithMetadata } from '../minio'
+import type { BucketItemFromList } from '../minio'
```
Comment (trim21) [inline src/internal/xml-parser.ts:7]: Maybe should move this type to internal/type.ts

File: src/internal/xml-parser.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { XMLParser } from 'fast-xml-parser'
 
 import * as errors from '../errors.ts'
 import type { BucketItemWithMetadata } from '../minio'
```
Comment (trim21) [inline src/internal/xml-parser.ts:6]: Also this one

**Comment by trim21** [comment]
Import type from minio.d.ts will cause resolution problem in dist files

File: src/internal/xml-parser.ts (hunk 4 above)
Comment (prakashsvmx) [reply src/internal/xml-parser.ts]: :+1:  Yes. i have updated the type reference.

File: src/internal/xml-parser.ts (hunk 3 above)
Comment (prakashsvmx) [reply src/internal/xml-parser.ts]: looks like it is already part of `internal/type.ts`. so just update the import

File: examples/list-buckets.js (hunk 2 above)
Comment (prakashsvmx) [reply examples/list-buckets.js]: Thank you. I have updated.

File: docs/API.md [hunk 5]
```diff
@@ -203,10 +201,16 @@ bucketStream emits Object with the format:-
 **Example**
 
 ```js
-minioClient.listBuckets(function (err, buckets) {
-  if (err) return console.log(err)
-  console.log('buckets :', buckets)
-})
+const listBuckets = async () => {
```
Comment (trim21) [inline docs/API.md:204]: how about top level await in this file？

Comment (prakashsvmx) [reply docs/API.md]: yes. updated . thank you. 
i have added note to the specific method. we can update to be global once the complete migration is done.

**Comment by trim21** [comment]
lgtm

File: docs/API.md [hunk 6]
```diff
@@ -202,11 +200,17 @@ bucketStream emits Object with the format:-
 
 **Example**
 
+Please refer to: [list-buckets.mjs](..%2Fexamples%2Flist-buckets.mjs)
+
+_Note_: the below example relies on top level await.
```
Comment (trim21) [inline docs/API.md:205]: maybe a global note instead every code snippet?

Comment (prakashsvmx) [reply docs/API.md]: sure. moved little up.

**Review by kaankabalak — APPROVED** [review]
LGTM

**Review by trim21 — COMMENTED** [review]
LGTM

**Pull request merged**